subprojects/docs/src/samples/generated/cpp-application/README.adoc

```diff
@@ -0,0 +1,235 @@
+This guide demonstrates how to create a C++ application with Gradle using `gradle init`.
```

Comment:
⭕ Just to understand the plan, this PR moves the guides to gradle/gradle and then, in an upcoming PR you (or somebody in the team) will integrate them into the `init` plugin and let the docs just generate the source.

Reply:
Yes. I plan to do this next week: https://github.com/gradle/gradle-private/issues/3186. Basically everything is already there on an experimental branch which I used to generate these file. But I have to move it to buildSrc and wire them correctly.

Then everything in the `generated` folder will be removed again and instead the templates (based on `init`) will be in the repo.

Just wanted to have this here as one incremental step. It's still a bit of technical work to wire everything up correctly.

Reply:
Sounds good, thanks for the explanation.

Reply:
This feels a bit hacky and imho should be fixed in the samples plugin. I'm fine with the change though. Could you please add a few line of the comments explaining this isssue?

Reply:
Great, thanks!